src/core/Akka.Persistence.Tests/Journal/ChaosJournal.cs

```diff
@@ -53,18 +55,18 @@ public ChaosJournal()
             _readHighestFailureRate = config.GetDouble("read-highest-failure-rate");
         }
 
-        public override Task ReplayMessagesAsync(string persistenceId, long fromSequenceNr, long toSequenceNr, long max, Action<IPersistentRepresentation> replayCallback)
+        public override Task ReplayMessagesAsync(IActorContext context, string persistenceId, long fromSequenceNr, long toSequenceNr, long max, Action<IPersistentRepresentation> replayCallback)
         {
+            var promise = new TaskCompletionSource<object>();
             if (ChaosSupportExtensions.ShouldFail(_replayFailureRate))
             {
                 var replays = Read(persistenceId, fromSequenceNr, toSequenceNr, max).ToArray();
-                var top = replays.Take(_random.Next(replays.Length + 1));
+                var top = replays.Take(_random.Next(replays.Length + 1)).ToArray();
```

Comment:
If we don't, we're going to enumerate multiple times since `top` is a param of `new ReplayFailedException(top)`. This code is only used in testing, so not really critical.